packages/salesforcedx-vscode-core/package.json

```diff
@@ -69,6 +69,12 @@
           "when": "view == sfdx.force.tasks.view"
         }
       ],
+      "explorer/context": [
+        {
+          "command": "sfdx.force.create.apex",
+          "when": "explorerResourceIsFolder"
```

Comment:
Follow the naming convention of `sfdx:explorer_resource_is_folder` for your when clause.